service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportController.java

```diff
@@ -29,13 +38,32 @@
 
     @PostMapping(value = "/report", consumes = { TEXT_XML_VALUE, APPLICATION_XML_VALUE })
     @ResponseStatus(value = ACCEPTED)
-    public void postReport(@RequestBody String reportXml) throws JsonProcessingException {
+    public void postReport(@RequestBody String reportXml) {
         try {
-            POCDMT000002UK01ClinicalDocument1 clinicalDocument = extractClinicalDocument(reportXml);
+            SAXReader reader = new SAXReader();
```

Comment:
Can we extract the parsing part to a separate class? Keep the controller as clean and simple as possible?

Reply:
done